Sources/ArgumentParser/Parsing/ArgumentSet.swift

```diff
@@ -394,7 +402,7 @@ extension ArgumentSet {
     // We have parsed all non-positional values at this point.
     // Next: parse / consume the positional values.
     do {
-      try parsePositionalValues(from: positionalValues, into: &result)
+      try parsePositionalValues(positionalValues, unusedOptions: unusedOptions, into: &result)
```

Comment:
I’m not sure, but: Would it make sense to interleave `positionalValues` and `unusedOptions` to get the original ordering? Possibly even use the `SplitArguments` type for this, since it already is aware of “a list of things that we parsed and their index into the original input array”…? I’m not sure.

Reply:
I ended up keeping the separate arrays, but changing the way they're consumed in `parsePositionalValues`. We can look at this from a performance standpoint later, if it's a problem.